multiqc/modules/verifybamid/verifybamid.py

```diff
@@ -0,0 +1,238 @@
+        self.verifybamid_table()
+
+
+    def parse_selfsm(self, f):
+        """ Go through selfSM file and create a dictionary with the sample name as a key, """
+        #create a dictionary to populate from this sample's file
+        parsed_data = dict()
+        # set a empty variable which denotes if the headers have been read
+        headers = None
+        # for each line in the file
+        for l in f['f'].splitlines():
+                # split the line on tab
+                s = l.split("\t")
+                # if we haven't already read the header line
+                if headers is None:
+                    # remove the leading hash from the first element of the header
+                    s[0] = s[0].lstrip('#')
+                    # assign this list to headers variable
+                    headers = s
+                # for all rows after the first
+                else:
+                    s_name=s[0]
+                    s_name = self.clean_s_name(s_name, f['root'])
+                    # create a dictionary entry with the first column as a key (sample name) and an empty dictionary as a value
+                    parsed_data[s_name] = {}
+                    # for each item in list of items in the row
+                    for i, v in enumerate(s):
+                    # if it's not the first element (if it's not the name)
+                        if i != 0:
+                            # try and convert the value into a float
+                            try:
+                            # and add to the dictionary the key as the corrsponding item from the header and the value from the list
+                                parsed_data[s_name][headers[i]] = float(v)
+                                #if can't convert to float...
+                            except ValueError:
+                            # add to the dictionary the key as the corrsponding item from the header and the value from the list
+                                parsed_data[s_name][headers[i]] = v
+    
+        # nothing has been parsed return None
+        if len(parsed_data) == 0:
+            return None
+        # else return the dictionary
+        return parsed_data
+
+    def verifybamid_general_stats_table(self):
+        """ Take the percentage of contamination from all the parsed *.SELFSM files and add it to the basic stats table at the top of the report """
+
+        # create a dictionary to hold the columns to add to the general stats table
+        headers = OrderedDict()
+        # available columns are:
+        #SEQ_ID RG  CHIP_ID #SNPS   #READS  AVG_DP  FREEMIX FREELK1 FREELK0 FREE_RH FREE_RA CHIPMIX CHIPLK1 CHIPLK0 CHIP_RH CHIP_RA DPREF   RDPHET  RDPALT
+        #,see https://genome.sph.umich.edu/wiki/VerifyBamID#Interpreting_output_files
+
+        # add the CHIPMIX column. set the title and description
+        headers['CHIPMIX'] = {
+            'title': 'Contamination Prediction - sequence and array',
```

Comment:
This is too long - it makes the column really wide, which is especially bad in the General Statistics table as it tends to be crowded with lots of columns.. Perhaps `Contamination S+A` or something? Better still with a shorter word alternative to _Contamination_, but I can't think of one..